11_notify/n0npax/pkg/puppy/rest.go

```diff
@@ -0,0 +1,132 @@
+package puppy
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	middleware "github.com/anz-bank/go-course/11_notify/n0npax/pkg/middleware"
+	"github.com/gin-gonic/gin"
+)
+
+var (
+	LostPuppyURL        = ""
+	SlowRequestDuration = "2s"
+	PuppyDeleteNotifyF  = LostPuppyReq
+)
+
+// LostPuppyReq sends request to lostpuppy service
+func LostPuppyReq(id int) {
+	jsonBytes := []byte(fmt.Sprintf(`{"id":%d}`, id))
+	resp, err := http.Post(LostPuppyURL, "application/json", bytes.NewBuffer(jsonBytes)) // #nosec
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	defer resp.Body.Close()
+	log.Printf("lostpuppy service response code: %d (puppy id: %d)", resp.StatusCode, id)
+}
+
+// LostPuppyBackend provides information if puppy was lost
+func LostPuppyBackend() *gin.Engine {
```

Comment:
IMO LostPuppyBackend should go in a separate file as it's used in a different process.

Reply:
fixed.